### PR TITLE
Use exponential backoff when downloading files

### PR DESCRIPTION
**Pull request by hmusum**

### CODE REVIEW

File: filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceDownloader.java [hunk 1]
```diff
@@ -57,12 +57,15 @@ private void startDownload(Duration timeout, FileReferenceDownload fileReference
...
+        int retryCount = 0;
         while ((System.currentTimeMillis() < end) && !downloadStarted) {
             try {
                 if (startDownloadRpc(fileReference)) {
                     downloadStarted = true;
                 } else {
-                    Thread.sleep(10);
+                    // Simple exponential backoff, might consider adding some randomness
+                    retryCount++;
+                    Thread.sleep(((int) Math.min(Math.round(Math.pow(2, retryCount)), 300) * 1000));
```
Comment (baldersheim) [inline filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceDownloader.java]: Which thread pool is this.

Comment (hmusum) [reply filedistribution/src/main/java/com/yahoo/vespa/filedistribution/FileReferenceDownloader.java]: This is a thread pool in the FileDownloader (fixed, 10 threads) used by both config proxy and config server

**Comment by yngveaasheim** [comment]
Is this due to exhausting the network?

**Comment by hmusum** [comment]
Discussed with Henning, we will rather use a longer timeout, to avoid many requests going to server and generating a unnecessary work server-side when answering the requests. Replaced by https://github.com/vespa-engine/vespa/pull/5016

**Pull request closed without merging**